services/web/server/src/simcore_service_webserver/login/handlers.py

```diff
@@ -149,6 +141,14 @@ async def login(request: web.Request):
     # user logs in
     identity = user["email"]
     response = flash_response(cfg.MSG_LOGGED_IN, "INFO")
+
+    # Store useful user information in the encrypted session
+    session = await new_session(request)
+    session['user'] = {
```

Comment:
so you are creating yet another cookie with non-encrypted info??
Why dont' you use the security session for that as what you did before?
Do you destroy the session at logout?
PLease add a test of this change . I think it is important enough

Reply:
everything in the session goes encrypted and inside just one cookie called... API_SESSION I think. when a user logs in, a new session should be created. calling ``remember`` and ``forget`` should deal with the session already

Reply:
No, IMO that is not accurate. A session object is like a dict and ``remember`` and ``forget`` creates and delete ONLY one item of that dict, specifically ``session["AIOHTTP_SECUTITY"]`` but not others.


Trying this workflow in a test:

- GET /v0/
- check session is empty
- a user logs in
- check session is not empty
- a user logs out
- check session is empty

as a reference, see fixtures used create a logged in user ... in ``services/web/server/tests/unit/with_dbs/test_login.py``

- @sanderegg did you add some mechanism to control the lifetime of a session? I see that logout triggers a signal but not sure if a garbare
- tip: could perhaps have a ``forget_all`` umbrella function that deletes all session

Reply:
> No, IMO that is not accurate. A session object is like a dict and `remember` and `forget` creates and delete ONLY one item of that dict, specifically `session["AIOHTTP_SECUTITY"]` but not others.
> 
> Trying this workflow in a test:
> 
> * GET /v0/
> * check session is empty
> * a user logs in
> * check session is not empty
> * a user logs out
> * check session is empty
> 
> as a reference, see fixtures used create a logged in user ... in `services/web/server/tests/unit/with_dbs/test_login.py`
> 
> * @sanderegg did you add some mechanism to control the lifetime of a session? I see that logout triggers a signal but not sure if a garbare
> * tip: could perhaps have a `forget_all` umbrella function that deletes all session

awesome, thanks!
then inside that one there is the identity, that is, the user email... I could use that directly. Then a good test would be that the identity is always the email. Or we can just add this dict and be sure that the email is always there, no matter what the identity is.
There is no need to destroy the session right? we can just create a new one at login time @pcrespov

Reply:
for the moment, I invalidated the session at logout, and added the user_email in the session also when calling /me

Reply:
yep. OK. 

> what would you consider meaningful? it is just one very simple function very likely to change in the short term!

I suggested some above

Reply:
> No, IMO that is not accurate. A session object is like a dict and `remember` and `forget` creates and delete ONLY one item of that dict, specifically `session["AIOHTTP_SECUTITY"]` but not others.
> 
> Trying this workflow in a test:
> 
> * GET /v0/
> * check session is empty
> * a user logs in
> * check session is not empty
> * a user logs out
> * check session is empty
> 
> as a reference, see fixtures used create a logged in user ... in `services/web/server/tests/unit/with_dbs/test_login.py`
> 
> * @sanderegg did you add some mechanism to control the lifetime of a session? I see that logout triggers a signal but not sure if a garbare
> * tip: could perhaps have a `forget_all` umbrella function that deletes all session

done